src/chrome/internal/locations/location.ts

```diff
@@ -0,0 +1,157 @@
+import * as Validation from '../../../validation';
+import { IScript } from '../scripts/script';
+import { ISource } from '../sources/source';
+import { ILoadedSource } from '../sources/loadedSource';
+import { logger } from 'vscode-debugadapter';
+import { ColumnNumber, LineNumber, URLRegexp } from './subtypes';
+import { CDTPScriptUrl } from '../sources/resourceIdentifierSubtypes';
+import { IResourceIdentifier, parseResourceIdentifier, URL } from '../sources/resourceIdentifier';
+
+export type integer = number;
+
+export class Coordinates {
+    public isSameAs(location: Coordinates): boolean {
+        return this.lineNumber === location.lineNumber
+            && this.columnNumber === location.columnNumber;
+    }
+
+    public toString(): string {
+        return this.columnNumber !== undefined
+            ? `${this.lineNumber}:${this.columnNumber}`
+            : `${this.lineNumber}`;
+    }
+
+    constructor(
+        public readonly lineNumber: LineNumber,
+        public readonly columnNumber?: ColumnNumber) {
+        Validation.zeroOrPositive('Line number', lineNumber);
+        if (columnNumber !== undefined) {
+            Validation.zeroOrPositive('Column number', columnNumber);
+        }
+    }
+}
+
+interface ILocation<T extends ScriptOrSourceOrURLOrURLRegexp> {
```

Comment:
What is this? It has coordinates and another line/col?

Reply:
The way I'm using these they are just an alias of the coordinates...
If you feel strongly about it, I can remove them and leave the coordinates only...

Reply:
Oh, ok. I think that's not really obvious from the interface. I think the interface should either expose the line/col or the coordinates, but not both. I don't really care which way.

Reply:
Just to confirm: Do you also want to remove them from the classes? Or just the interface? (I'd like to leave them on the classes)

Reply:
Leaving them on the class as a private thing is ok. But the interface should make it clear how it's supposed to be used.

Reply:
I'll remove them from the classes then. Thanks!